### ISSUE TITLE
Connection over TLS affects default stream context

### ISSUE THREAD

**Issue by esler** [predis/predis#742]
Socket connection is opened by `stream_socket_client()` without context. PHP will assign default (global) context to the resource. Later in the connection init functions like `stream_context_set_option()` will change resource's context which's global and shared by all file handling functions. That may lead to some nasty issues including security vulnerabilities. I already prepared PR (#743) with a fix.

1. create a `tls` connection with custom certificate
2. fetch someting from valid `https` later in the code
3. see code for more details

Library should not affect any subsequent connections

 - Predis: 1.1.10 (but affects almost all versions, it's there for 7 years)
 - PHP 7.4 and 8.1 (but all supported versions are affected)
 - Redis Server 6.0 (any with TLS support)
 - OS Linux, MacOS

```php
$redis = new Client([
  'scheme' => 'tls',
  'ssl' => [
    'cafile' => 'redis.crt', // self-signed
  ],
]);

copy(
  'https://www.wikipedia.org/portal/wikipedia.org/assets/img/Wikipedia-logo-v2.png',
  '/tmp/wiki-logo.png'
);
// PHP Warning:  copy(): SSL operation failed with code 1. OpenSSL Error messages: 
// error:1416F086:SSL routines:tls_process_server_certificate:certificate verify failed in ...
```

**Comment by esler** [comment]
#714 is most probably caused by this issue

**Comment by tillkruss** [comment]
Fixed in #743.
